src/types/UiaTextRangeBase.cpp

```diff
@@ -1013,7 +1013,9 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         }
     }
 
-    if (maxLength >= 0)
+    // if the caller asked for INT_MAX,
+    // they probably just want the whole thing
+    if (maxLength >= 0 && maxLength != INT_MAX)
     {
         textData.resize(maxLength);
     }

```

Comment:
@carlos-zamora Correct me if I'm wrong, but to me it sounds like we're supposed to only shrink the string... But `resize()` with a `maxLength` greater than the string length pads it with `L'\0'`s at the end. Do we want that?

```suggestion
    if (const auto s = gsl::narrow_cast<size_t>(maxLength); s < textData.size())
    {
        textData.resize(s);
    }
```

Reply:
huh, that's probably _why_ the resize takes so long. Good catch.